src/fx/AnimateableTimeStretchPitchScaleReader.cpp

```diff
@@ -0,0 +1,62 @@
+
+
+/*******************************************************************************
+ * Copyright 2009-2025 Jörg Müller
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+#include "fx/AnimateableTimeStretchPitchScaleReader.h"
+
+#include "IReader.h"
+
+AUD_NAMESPACE_BEGIN
+
+float getValueAtPosition(std::shared_ptr<AnimateableProperty> prop, float position)
+{
+	float value;
+	prop->read(position, &value);
+	return value;
+}
```

Comment:
You could mark this function `static` (to hide it in the compilation unit) and `inline` (to mark it for optimization) or - going the other direction (making it more accessible) - you could add a `readSingle` method to `AnimateableProperty` (ideally with with an assert on `m_count` to ensure it's 1). Either way, I'd also use it in the read and seek methods.